pkg/token/daemon.go

```diff
@@ -143,26 +145,100 @@ func newDaemon(idConfig *config.IdentityConfig, tt mode) (*daemon, error) {
 	}, nil
 }
 
-func (d *daemon) updateTokenWithRetry() error {
+func (d *daemon) updateTokenCaches() <-chan error {
+	var atErrorCount, rtErrorCount atomic.Int64
+	atTargets := d.accessTokenCache.Keys()
+	rtTargets := d.roleTokenCache.Keys()
+	log.Infof("Attempting to fetch tokens from Athenz ZTS server: access token targets[%v], role token targets[%v]...", atTargets, rtTargets)
+	echan := make(chan error, len(atTargets)+len(rtTargets))
+	go func() {
```

Comment:
```suggestion
```
no need for `go func() {`, just let the function to be blocking.

`for err := range d.updateTokenCaches() {` will always block until the channel close.

Reply:
fixed: https://github.com/AthenZ/k8s-athenz-sia/pull/75/commits/24f9c761a3156502844efccc31dbfaa3cd52a193